### DISCUSSION TITLE
How can i fix this? X-axis problems in line and bar graphs

### DISCUSSION THREAD

**Discussion by FrancoPaez** [plouc/nivo#1618] [Q&A] [answered]
I really don't even know how to look for this problem. What I want is for the X axis to stay down when there is no data, not in the middle.

![a](https://user-images.githubusercontent.com/82821596/123125897-516b6700-d41f-11eb-966f-a01aeebfa8c0.gif)

It should look like the figma

Line chart
![asdf](https://user-images.githubusercontent.com/82821596/123126589-e1111580-d41f-11eb-9d6c-51c97c96f84d.PNG)

Bar chart
![asdw](https://user-images.githubusercontent.com/82821596/123127034-495ff700-d420-11eb-8865-dc4d252721fe.PNG)

I have this problem in both line and bar charts

**Comment by wyze** [comment, marked as answer]
You can implement this with custom layer and providing a `min` & `max` value for the `yScale`/`valueScale` props.

Here is an example with Line chart, you can do something similar for the Bar. https://codesandbox.io/s/interesting-visvesvaraya-ncl5o?file=/src/App.tsx

**Reply by FrancoPaez** [reply]
> You can implement this with custom layer and providing a `min` & `max` value for the `yScale`/`valueScale` props.
> 
> Here is an example with Line chart, you can do something similar for the Bar. https://codesandbox.io/s/interesting-visvesvaraya-ncl5o?file=/src/App.tsx

It is a good solution, but the problem is that I never know the maximum of my graph, it can be 3 or 300.

If I put a very low number, the graph cuts out.

and if I put a very large value, the graph looks small.

would there be no other solution?

**Reply by wyze** [reply]
If there is no data to pass to the scale, it is hard to determine where the ticks should go. Currently, nivo doesn't have any other solution. If you can find something another library does, we could draw some inspiration from it and see if we can achieve similar results.

**Reply by FrancoPaez** [reply]
okay, thank you very much!

**Reply by plouc** [reply]
@FrancoPaez, you could store the latest max value using a hook, then if you receive an update, and there's no data, you can use it to set the max value of the scale, this seems a bit too specific to be added to the lib itself IMHO.

**Reply by FrancoPaez** [reply]
I agree, I was just wondering if there was an option in the library for that. Thanks!
